modules/planning/scenarios/scenario_manager.cc

```diff
@@ -461,10 +461,20 @@ ScenarioConfig::ScenarioType ScenarioManager::SelectStopSignScenario(
 
 ScenarioConfig::ScenarioType ScenarioManager::SelectTrafficLightScenario(
     const Frame& frame, const hdmap::PathOverlap& traffic_light_overlap) {
-  // scenario conf is the same across all traffic-light scenarios
-  const auto& scenario_config =
-      config_map_[ScenarioConfig::TRAFFIC_LIGHT_PROTECTED]
-          .traffic_light_protected_config();
+
+  // some scenario may need start sooner than the others
+  const double start_check_distance =
+      std::max(
+          std::max(
+              config_map_[ScenarioConfig::TRAFFIC_LIGHT_PROTECTED]
+                              .traffic_light_protected_config()
+                              .start_traffic_light_scenario_distance(),
+              config_map_[ScenarioConfig::TRAFFIC_LIGHT_UNPROTECTED_LEFT_TURN]
+                              .traffic_light_unprotected_left_turn_config()
+                              .start_traffic_light_scenario_distance()),
+          config_map_[ScenarioConfig::TRAFFIC_LIGHT_UNPROTECTED_RIGHT_TURN]
+                          .traffic_light_unprotected_right_turn_config()
+                          .start_traffic_light_scenario_distance());
```

Comment:
Could you try `std::max({a, b, c})`?
https://en.cppreference.com/w/cpp/algorithm/max

Reply:
amended.  thx * n !!